File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    man1: list[int] = [1, 2, 3, 4, 5]
    man2: list[int] = [2, 1, 2, 3, 2, 4, 2, 5]
    man3: list[int] = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    scores: list[int] = [0, 0, 0]
    answer: list[int] = []
    for i, num in enumerate(answers):
        if num == man1[i % len(man1)]:
            scores[0] += 1
        if num == man2[i % len(man2)]:
            scores[1] += 1
        if num == man3[i % len(man3)]:
            scores[2] += 1
    maxi: int = max(scores)
    for i, num in enumerate(scores):
        if num == maxi:
            answer.append(i+1)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(numbers) {
    var num1 = [1, 2, 3, 4, 5];
    var num2 = [2, 1, 2, 3, 2, 4, 2, 5];
    var num3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    var numCount = [0, 0, 0];
    numbers.forEach(function (num, index) {
        if (num1[index % num1.length] === num) {
            numCount[0]++;
        }
        if (num2[index % num2.length] === num) {
            numCount[1]++;
        }
        if (num3[index % num3.length] === num) {
            numCount[2]++;
        }
    });
    var maxCount = Math.max.apply(Math, numCount);
    var answer = [];
    numCount.forEach(function (num, index) {
        if (num === maxCount) {
            answer.push(index+1);
        }
    });
    return answer;
}